stdlib/public/core/Arrays.swift.gyb

```diff
@@ -453,6 +453,29 @@ if True:
     raise ValueError('Unhandled case: ' + Self)
 }%
 
+@_fixed_layout
```

Comment:
Could this be a nested type of ${Self}, like @airspeedswift's other PR?

Reply:
Is this effectively a hand-coded specialization of IndexingIterator?

Reply:
Good call, will do

Reply:
Yes